packages/tools/src/utilities/math/polyline/getSignedArea.ts

```diff
@@ -0,0 +1,47 @@
+import type { Types } from '@cornerstonejs/core';
+
+/**
+ * Returns the area with signal of a 2D polyline
+ * https://www.youtube.com/watch?v=GpsKrAipXm8&t=1900s
+ *
+ * This functions has a runtime very close to `getArea` and it is recommended to
+ * be called only if you need the area signal (eg: calculate polygon normal). If
+ * you do not need the area signal you should always call `getArea`.
+ *
+ *
+ * @param polyline - Polyline points (2D)
+ * @returns Area of the polyline (with signal)
+ */
+export default function getSignedArea(polyline: Types.Point2[]): number {
```

Comment:
After refactoring the function and removing any `glMatrix` function call it is 2-5x faster (2x faster  for 1k points, 4x faster for 7k points, 5x faster for 20k points) and it is <4% slower than `getArea` (it was faster ~15% of the time when some tests)